### PR TITLE
Make `make tar' ignore DESTDIR

### PR DESCRIPTION
**Pull request by satan1st**
### Description of your *pull request* and other information

DESTDIR is only usefull in install* and uninstall* target[0]. The tar
target only generates a distribution file and doesn't install anything.

Also don't set DESTDIR if not set.

[0] https://www.gnu.org/prep/standards/html_node/DESTDIR.html

Direct installing from the a tar or the git repo currently requires to manual unset DESTDIR.

Fixes #

<details open><summary>Template</summary>

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) No python code changed.

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))
- [x] Build bug fix/improvement

</details>

### CODE REVIEW

**Comment by pukkandan** [comment]
This was added in https://github.com/yt-dlp/yt-dlp/pull/129/commits/efa5501dd3b0403e80c0d4780e2101b6bcbd410f 

c @hseg

**Comment by hseg** [comment]
Hrm. I see I explicitly made this change, but cannot recall why I thought it correct - presumably I considered this another `install`-style target.
Considering this is a synonym for `make dist`, that's probably mistaken, as noted in the gnu manual linked, though I don't know why - am looking up.
The setting of `DESTDIR` is definitely in error judging by that manpage - though given that that is afaict the default, I'm not sure what the impact is.

Tldr: Looks like the PR is technically correct, am looking into it, am unsure what the harm of status quo is given that it essentially replicates defaults.

Am 31. August 2022 20:28:23 GMT+03:00 schrieb pukkandan ***@***.***>:

**Comment by satan1st** [comment]
As a bit context: DESTDIR is for packaging systems, to "install" a program in a fake root directory. But in the normal case it is unset so you can run `make install` to install it on your system or run `make install DESTDIR=/path/to/build/output` and make an archive out of the directory.

I have made this pr exactly because of this. I install yt-dlp directly from the git repo and manual unsetting DESTDIR bugs me a bit. So I have look a bit for the reasons and found #129 and https://github.com/yt-dlp/yt-dlp/issues/3429#issuecomment-1106879598. But no explanation why this is set up in such uncommon way.
